packages/amesos2/test/adapters/Tpetra_CrsMatrix_Adapter_UnitTests.cpp

```diff
@@ -60,6 +60,24 @@
 #include "Amesos2_MatrixAdapter_def.hpp"
 #include "Amesos2_Meta.hpp"
 
+namespace TestTraitsNS {
```

Comment:
I might have preferred an approach where the matrix file comes in through a command-line argument.  That would make it possible to run the test executable anywhere, not just in the build or install directory.  This is fine, though :-)

Reply:
Also, instead of partial specialization, you could use type traits:
```
if (Teuchos::ScalarTraits<Scalar>::isComplex) {
  return "../matrices/..amesos2_test_mat0_complex.mtx";
}
else {
  return "../matrices/..amesos2_test_mat0.mtx";
} 
```

Reply:
This is fine, though; I don't mean to hold up the fix.